src/machine/state/mem.rs

```diff
@@ -2,67 +2,208 @@ use machine::state::types::*;
 
 use std::ops::{Index, IndexMut};
 
+#[derive(Copy, Clone)]
 pub struct ByteAt(pub u64);
+
+#[derive(Copy, Clone)]
 pub struct WydeAt(pub u64);
+
+#[derive(Copy, Clone)]
 pub struct TetraAt(pub u64);
+
+#[derive(Copy, Clone)]
 pub struct OctaAt(pub u64);
 
-pub struct Memory(());
+pub struct Memory {
+    buf: Vec<Octa>,
+}
 
 impl Memory {
     pub fn new() -> Self {
         unimplemented!();
     }
+
+    pub fn with_capacity(capacity: u64) -> Self {
+        // Compute the number of octas needed
+        let len = {
+            let mut len = (capacity / 8) as usize;
+            if capacity % 8 > 0 {
+                len += 1;
+            }
+            len
+        };
```

Comment:
Maybe rename `capacity` to `bytes` to make the unit more clear? Could also use `(capacity + 7) / 8` instead.